csharp/src/Google.Protobuf/Collections/RepeatedFieldExtensions.cs

```diff
@@ -0,0 +1,53 @@
+#region Copyright notice and license
+// Protocol Buffers - Google's data interchange format
+// Copyright 2015 Google Inc.  All rights reserved.
+//
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file or at
+// https://developers.google.com/open-source/licenses/bsd
+#endregion
+
+using System;
+
+namespace Google.Protobuf.Collections
+{
+    /// <summary>
+    /// A set of extension methods on RepeatedField
+    /// </summary>
+    public static class RepeatedFieldExtensions
+    {
+        /// <summary>Adds the elements of the specified span to the end of the <see cref="RepeatedField{T}"/>.</summary>
+        /// <typeparam name="T">The type of elements in the <see cref="RepeatedField{T}"/>.</typeparam>
+        /// <param name="repeatedField">The list to which the elements should be added.</param>
+        /// <param name="source">The span whose elements should be added to the end of the <see cref="RepeatedField{T}"/>.</param>
+        /// <exception cref="ArgumentNullException">The <paramref name="repeatedField"/> is null.</exception>
+        public static void AddRange<T>(this RepeatedField<T> repeatedField, ReadOnlySpan<T> source)
```

Comment:
Change this to an overload on RepeatedField.

.NET added `AddRange` as an extension method to `List<T>` to avoid AOT file growth. RepeatedField has much narrower usage and that isn't a concern.

Using an overload also eliminates need to make some fields internal on RepeatedField.

Reply:
AddRange is added as an extension method instead of directly adding it to RepeatedField<T> because we want to avoid ambiguity between RepeatedField<T>.AddRange(IEnumerbale<T>) and RepeatedField<T>.AddRange(Span<T>) for types such as T[]. This is the same exact approach taken by .NET runtime itself.
I agree that AOT is not a concern here.